src/redux/paymentMethodsSlice.js

```diff
@@ -0,0 +1,70 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { getPaymentMethods } from '../api';
+import submitPaymentWithoutDetails from '../api/Payment/submitPaymentWithoutDetails';
+
+const initialState = {
+  paymentMethods: null,
+  loading: false,
+  error: null,
+  availablePaymentMethods: null
+};
+
+export const fetchPaymentMethods = createAsyncThunk(
+  'paymentMethods/fetchPaymentMethods',
+  async () => {
+    const { paymentResponse } = await getPaymentMethods();
+    return paymentResponse;
+  }
+);
+
+export const fetchPaymentWithoutDetails = createAsyncThunk(
+  'paymentMethods/fetchPaymentWithoutDetails',
+  async () => {
+    const {
+      responseData: { paymentMethods }
```

Comment:
this call will give you back a payment object, not paymentMethods

Reply:
need more information